src/bijectors/corr.jl

```diff
@@ -0,0 +1,118 @@
+# See stan doc for parametrization method:
+# https://mc-stan.org/docs/2_23/reference-manual/correlation-matrix-transform-section.html
+# (7/30/2020) their "manageable expression" is wrong...
+
+struct CorrBijector <: Bijector{2} end
+
+function (b::CorrBijector)(x::AbstractMatrix{<:Real})    
+    w = cholesky(x).U + zero(x) # convert to dense matrix
+    r = _link_chol_lkj(w) 
+    return r
+end
+
+(b::CorrBijector)(X::AbstractArray{<:AbstractMatrix{<:Real}}) = map(b, X)
+
+function (ib::Inverse{<:CorrBijector})(y::AbstractMatrix{<:Real})
+    w = _inv_link_chol_lkj(y)
+    return w' * w
+end
+(ib::Inverse{<:CorrBijector})(Y::AbstractArray{<:AbstractMatrix{<:Real}}) = map(ib, Y)
+
+
+function logabsdetjac(::Inverse{CorrBijector}, y::AbstractMatrix{<:Real})
+    @assert size(y, 1) == size(y, 2)
+    K = size(y, 1)
+    
+    left = zero(eltype(y)) # Initial summand may make looping looks weird :(
+    @inbounds for j=2:K, i=1:(j-1)
+        # left += (K-i-1) * log(1 - tanh(y[i, j])^2) # lacks numerically stable
+        left += (K-i-1) * 2 * (log(2) - log(exp(y[i,j]) + exp(-y[i,j])))
+    end
+    
+    right = zero(eltype(y))
+    @inbounds for j=2:K, i=1:(j-1)
+        right += log(cosh(y[i, j])^2)
+    end
+    
+    return left / 2 - right
```

Comment:
It seems the multiplication with 2 here cancels with the division by 2 below? Additionally, the computation of `right` can also be simplified to `2 * (-log(2) + log(exp(-y[i, j]) + exp(y[i, j]))`. Taken together take makes `(K - i + 1) * (log(2) - log(exp(y[-y[i, j]) + exp(y[i, j])))` for each combination of `i` and `j`. Therefore I'd suggest I'd suggest
```suggestion
    result = float(zero(eltype(y))
    for j in 2:K, i in 1:(j - 1)
        @inbounds abs_y_i_j = abs(y[i, j])
        result += (K - i + 1) * (logtwo - (abs_y_i_j + log1pexp(-2 * abs_y_i_j)))
    end
    
    return result
```

Reply:
It seems that `abs` is not required? Let x = y[i,j]

(K-i-1) * (log(2) - log(exp(x) + exp(-x))) - log(cosh(y[i, j])^2)
(K-i-1) * (log(2) - log(2cosh(x))) - 2log(abs(cosh(x)))
(K-i-1) * (log(2) - log(2) - log(cosh(x))) - 2log(cosh(x))
(K-i+1) * (-log(cosh(x)))
(K-i+1) * ( -log( (1+exp(-2x))/(2exp(-x)) ) )
(K-i+1) * ( -log(1+exp(-2x)) + log(2exp(-x)))
(K-i+1) * (-log1pexp(-2x) + logtwo -x)
(K-i+1) * (logtwo - (x + log1pexp(-2x)))

Reply:
Mathematically it doesn't matter if you move `x` or `-x` out of the logarithm but for increased numerical stability it should be the larger of both terms, i.e. `abs(x)` (as done e.g. also in the implementation of StatsFuns.logaddexp).

Reply:
I see, it's useful when `-2x` is positive.